### PR TITLE
CVSL-274: Save COM details on licence submission

### PR DESCRIPTION
**Pull request by smcveigh941**
- COM details have before been saved at the point of creation. They should now be saved at the point the licence is submitted.
- The username of the person who created the licence will still be recorded

### CODE REVIEW

File: server/routes/creatingLicences/handlers/caseload.ts [hunk 1]
```diff
@@ -28,9 +29,8 @@ export default class CaseloadRoutes {
     const { user } = res.locals
     const { prisonerNumber } = req.body
 
-    // TODO: This block is temporary, remove this when we have a design for editing existing licences
-    const comLicenses = await this.licenceService.getLicencesByStaffIdAndStatus([], user)
-    const existingLicense = comLicenses.find(licence => licence.nomisId === prisonerNumber)
+    // TODO: Which statuses should be considered "existing"? Should INACTIVE and RECALLED be included?
+    const existingLicense = _.head(await this.licenceService.getLicencesByNomisIdsAndStatus([prisonerNumber], [], user))
     if (existingLicense) {
```
Comment (timharrison-moj) [inline server/routes/creatingLicences/handlers/caseload.ts:34]: Just spelling `licence` not `license` :-)

File: server/routes/creatingLicences/handlers/caseload.ts [hunk 2]
```diff
@@ -28,9 +29,8 @@ export default class CaseloadRoutes {
     const { user } = res.locals
     const { prisonerNumber } = req.body
 
-    // TODO: This block is temporary, remove this when we have a design for editing existing licences
-    const comLicenses = await this.licenceService.getLicencesByStaffIdAndStatus([], user)
-    const existingLicense = comLicenses.find(licence => licence.nomisId === prisonerNumber)
+    // TODO: Which statuses should be considered "existing"? Should INACTIVE and RECALLED be included?
+    const existingLicense = _.head(await this.licenceService.getLicencesByNomisIdsAndStatus([prisonerNumber], [], user))
```
Comment (timharrison-moj) [inline server/routes/creatingLicences/handlers/caseload.ts:33]: Possible that the first in the list is an old, ACTIVE licence for this offender, when the one we want is in progress? Might need some type of filter / choice - but ok for now.

File: server/services/caseloadService.ts [hunk 3]
```diff
@@ -22,12 +22,8 @@ export default class CaseloadService {
       .filter(offender => offender.nomsNumber)
       .map(offender => offender.nomsNumber)
 
-    /*
-     TODO: Maybe this should be checking for existing licences by nomisId rather than staffId? What if the offender
-      changes their managing officer after a licence has been created for them?
-     */
-
```
Comment (timharrison-moj) [inline server/services/caseloadService.ts:29]: Yay!

File: server/services/licenceService.test.ts [hunk 4]
```diff
@@ -41,7 +47,6 @@ describe('Licence Service', () => {
   describe('Create Licence', () => {
     beforeEach(() => {
       prisonerService.getPrisonerDetail.mockResolvedValue({} as PrisonApiPrisoner)
-      communityService.getStaffDetail.mockResolvedValue({} as CommunityApiStaffDetails)
       communityService.getProbationer.mockResolvedValue({ offenderManagers: [] } as OffenderDetail)
```
Comment (timharrison-moj) [inline server/services/licenceService.test.ts:50]: I have a suspicion that we'll need to com's details on the licence whilst it is still in progress, as OMU users (I think) might be getting visibility of IN_PROGRESS licences, and not be able to contact or see the COMs details until is it submitted. One for the team to discuss..

File: server/services/licenceService.ts [hunk 5]
```diff
@@ -85,13 +87,9 @@ export default class LicenceService {
       ]
         .filter(n => n)
         .join(' '),
-      comUsername: staffDetail.username,
-      comStaffId: staffDetail.staffIdentifier,
-      comEmail: staffDetail.email,
-      comFirstName: offenderManager?.staff?.forenames,
-      comLastName: offenderManager?.staff?.surname,
       probationAreaCode: offenderManager?.probationArea?.code,
```
Comment (timharrison-moj) [inline server/services/licenceService.ts:90]: As above - might need these are creation time so they are present whilst its IN_PROGRESS.

**Review by timharrison-moj — COMMENTED** [review]
LGTM. I would like to confirm with the team whether the COMs details need to be on the licence whilst it is IN_PROGRESS so that : 
* They can be notified of events that affected it
* The OMU users can see who is creating the licence.
If we can get clear answers to these questions, I'm happy with the changes.

File: server/services/licenceService.test.ts (hunk 4 above)
Comment (smcveigh941) [reply server/services/licenceService.test.ts:50]: I chatted to Jon about this, if it's still in progress we can pull it from delius, but thats out of scope for now

**Review by timharrison-moj — APPROVED** [review]
LGTM. We can see how it goes and whether COM details required earlier in the process.

File: server/routes/creatingLicences/handlers/caseload.ts (hunk 1 above)
Comment (smcveigh941) [reply server/routes/creatingLicences/handlers/caseload.ts:34]: You'd think I'd know that by now!

File: server/routes/creatingLicences/handlers/caseload.ts (hunk 2 above)
Comment (smcveigh941) [reply server/routes/creatingLicences/handlers/caseload.ts:33]: Its possible yes, left a TODO to consider the case where the offender already has a licence which is INACTIVE or RECALLED

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**